src/scripts/setup.sh

```diff
@@ -52,14 +56,23 @@ else
 	echo "GH CLI is already installed."
 fi
 
-# Authenticate
-echo
-echo "Authenticating GH CLI"
-git config --global credential.https://github.com.helper ''
-git config --global --add credential.https://github.com.helper '!'"$(which gh) auth git-credential"
-gh auth status
+# if GITHUB_HOSTNAME is set use login
+if [ -z "$GITHUB_HOSTNAME" ]; then
+	# Authenticate
+	echo
+	echo "Authenticating GH CLI"
+	git config --global credential.https://github.com.helper ''
+	git config --global --add credential.https://github.com.helper '!'"$(which gh) auth git-credential"
+	gh auth status
 
-# Configure
-echo
-echo "Disabling interactive prompts for GH CLI"
-gh config set prompt disabled
+	# Configure
+	echo
+	echo "Disabling interactive prompts for GH CLI"
+	gh config set prompt disabled
+else
+	# Login
+	echo
+	echo "Authenticating GH CLI with hostname"
+	gh auth login --hostname "$GITHUB_HOSTNAME"  --with-token "$GITHUB_TOKEN"
+	gh auth status
```

Comment:
Do we need to setup our credential git config here as well?

Reply:
This one I am unsure about. According to https://cli.github.com/manual/gh_auth_login it's not in their example usage. I'm looking into but would defer to your expertise. 😄

Reply:
this looks promising https://github.com/cli/cli/issues/3796

Reply:
I don't have a gh enterprise instance to test this against, but I'm going to tentatively say let's add this and remove it if it proves to be a problem @motionsuggests From the PR is looks like it would be useful for enterprise/non-github.com instances.

Reply:
Based on the closed issue reported in the CLI, is this still needed?